schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -614,6 +620,81 @@ private static String encode(String schemaName) {
     }
 
     @Override
+    public void enableSchemaVersion(Long schemaVersionId)
+            throws SchemaNotFoundException, SchemaLifecycleException, IncompatibleSchemaException {
+        try {
+            fetchSchemaVersionState(schemaVersionId, "enable");
+        } catch (SchemaLifecycleException e) {
+            Throwable cause = e.getCause();
+            if(cause != null && cause instanceof IncompatibleSchemaException) {
+                throw (IncompatibleSchemaException) cause;
+            }
+            throw e;
+        }
+    }
+
+    @Override
+    public void disableSchemaVersion(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "disable");
+    }
+
+    @Override
+    public void deleteSchemaVersion(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "delete");
+    }
+
+    @Override
+    public void archiveSchemaVersion(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "archive");
+    }
+
+    @Override
+    public void startSchemaVersionReview(Long schemaVersionId) throws SchemaNotFoundException, SchemaLifecycleException {
+        fetchSchemaVersionState(schemaVersionId, "startReview");
+
+    }
+
+    private boolean fetchSchemaVersionState(Long schemaVersionId,
+                                            String operation) throws SchemaNotFoundException, SchemaLifecycleException {
```

Comment:
on that note, it seems we switch between tinyint and string, should keep to one or the other?